src/cr/cube/dimension.py

```diff
@@ -507,14 +506,28 @@ def subtotals(self):
         if self.dimension_type in (DT.MR, DT.CA_SUBVAR):
             insertion_dicts = []
         # --- insertions in dimension-transforms override those on dimension itself ---
-        elif "insertions" in self._dimension_transforms_dict:
-            insertion_dicts = self._dimension_transforms_dict["insertions"]
+        elif "insertions" in self._shimmed_dimension_transforms_dict:
+            insertion_dicts = self._shimmed_dimension_transforms_dict["insertions"]
         # --- otherwise insertions defined on dimension/variable apply ---
         else:
-            view = self._dimension_dict.get("references", {}).get("view") or {}
+            view = self._shimmed_dimension_dict.get("references", {}).get("view") or {}
             insertion_dicts = view.get("transform", {}).get("insertions", [])
         return _Subtotals(insertion_dicts, self.valid_elements)
 
+    def translate_element_id(self, _id):
+        """Optional string that is the translation of various ids to subvariable alias
+
+        0) If dimension is not a subvariables dimension, return the _id.
+        1) If id matches an alias, then just use it.
+        2) If id matches a subvariable id, translate to corresponding alias.
+        3) If id matches an element id, translate to corresponding alias.
+        4) If id can be parsed to int and matches an element id, translate to alias.
+        5) If id is int (or can be parsed to int) and can be used as index (eg in range
+           0-# of elements), use _id'th alias.
+        6) If all of these fail, return None.
+        """
+        return self._element_id_shim.translate_element_id(_id)
```

Comment:
I'm not super happy about it, but it's for the opposing element sort by value transform. This shimmed dimension transform dictionary does not have a translated element id here, because to translate it would require the dimension to understand it's opposing dimension at creation time, which seemed hard to manage.

Added the following to the docstring.

>        This is needed for the opposing dimension's sort by opposing element, because
>        when creating a dimension, we don't have access to the other dimension's 
>        ids to transform it. Therefore, the id for opposing element sort by value
>        transforms is not translated at creation time.